src/compiler/parser.ts

```diff
@@ -1390,10 +1404,23 @@ module ts {
                 return nextToken() === SyntaxKind.EnumKeyword;
             }
 
+            if (token === SyntaxKind.AsyncKeyword) {
+                nextToken();
+                return canFollowAsyncModifier();
+            }
+
             nextToken();
             return canFollowModifier();
         }
 
+        function canFollowAsyncModifier(): boolean {
+            if (scanner.hasPrecedingLineBreak()) {

```

Comment:
Should I remove the preceedingLineBreak check? We discussed this a bit before the holidays.